modules/l4http/httpmatcher.go

```diff
@@ -78,11 +80,18 @@ func (m MatchHTTP) Match(cx *layer4.Connection) (bool, error) {
 	req, ok := cx.GetVar("http_request").(*http.Request)
 	if !ok {
 		var err error
-		bufReader := bufio.NewReader(cx)
+
+		data := cx.MatchingBytes()
+		if !m.isHttp(data) {
```

Comment:
No it also matches http2 (see [TestMatchHTTP_isHttp](https://github.com/mholt/caddy-l4/pull/192/files#diff-0f169956149b8041f34a0b167acc8d7cdcbe68329fdb36cfd333c9f403ae7e79R273)), since http2 requests start http1 parser friendly/compatible.